agent/grpc-external/stats_test.go

```diff
@@ -0,0 +1,130 @@
+package external
+
+import (
+	"context"
+	"net"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/consul/agent/grpc-middleware/testutil"
+	"github.com/hashicorp/consul/agent/grpc-middleware/testutil/testservice"
+	"github.com/hashicorp/consul/proto/prototest"
+)
+
+func TestServer_EmitsStats(t *testing.T) {
+	sink, reset := patchGlobalMetrics(t)
```

Comment:
What do you think of passing the metrics object as a parameter to `NewServer` so we don't have to change global state like this?

```Go
func NewServer(logger agentmiddleware.Logger, metrics *metrics.Metrics) *grpc.Server {
	if metrics == nil {
		metrics = agentmiddleware.DefaultMetrics
	}
```

Reply:
I also appreciate that these tests were copied from `grpc-internal` so please feel free to leave as-is! 🙇🏻‍♂️